exporter/sumologicexporter/prometheus_formatter.go

```diff
@@ -0,0 +1,445 @@
+	dps := record.metric.DoubleSum().DataPoints()
+	lines := make([]string, 0, dps.Len())
+
+	for i := 0; i < dps.Len(); i++ {
+		dp := dps.At(i)
+		line := f.doubleDataPointValueLine(
+			record.metric.Name(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+	}
+
+	return lines
+}
+
+// doubleSummary2Strings converts DoubleSummary record to list of strings
+// for each data points: one per quantile plus two for sum and count
+func (f *prometheusFormatter) doubleSummary2Strings(record metricPair) []string {
+	dps := record.metric.DoubleSummary().DataPoints()
+	var lines []string
+
+	for i := 0; i < dps.Len(); i++ {
+		dp := dps.At(i)
+		qs := dp.QuantileValues()
+		additionalAttributes := pdata.NewAttributeMap()
+		for i := 0; i < qs.Len(); i++ {
+			q := qs.At(i)
+			additionalAttributes.UpsertDouble("quantile", q.Quantile())
+
+			line := f.doubleValueLine(
+				record.metric.Name(),
+				q.Value(),
+				dp,
+				f.mergeAttributes(record.attributes, additionalAttributes),
+			)
+			lines = append(lines, line)
+		}
+
+		line := f.doubleValueLine(
+			f.sumMetric(record.metric.Name()),
+			dp.Sum(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+
+		line = f.uintValueLine(
+			f.countMetric(record.metric.Name()),
+			dp.Count(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+	}
+	return lines
+}
+
+// intHistogram2Strings converts IntHistogram record to list of strings
+// for each data points: one per bound (with one for infinity) plus two for sum and count
+func (f *prometheusFormatter) intHistogram2Strings(record metricPair) []string {
+	dps := record.metric.IntHistogram().DataPoints()
+	var lines []string
+
+	for i := 0; i < dps.Len(); i++ {
+		dp := dps.At(i)
+
+		explicitBounds := dp.ExplicitBounds()
+		var cumulative uint64
+		additionalAttributes := pdata.NewAttributeMap()
+
+		for i, bound := range explicitBounds {
+			cumulative += dp.BucketCounts()[i]
+			additionalAttributes.UpsertDouble("le", bound)
+
+			line := f.uintValueLine(
+				record.metric.Name(),
+				cumulative,
+				dp,
+				f.mergeAttributes(record.attributes, additionalAttributes),
+			)
+			lines = append(lines, line)
+		}
+
+		cumulative += dp.BucketCounts()[len(explicitBounds)]
+		additionalAttributes.UpsertString("le", "+Inf")
+		line := f.uintValueLine(
+			record.metric.Name(),
+			cumulative,
+			dp,
+			f.mergeAttributes(record.attributes, additionalAttributes),
+		)
+		lines = append(lines, line)
+
+		line = f.intValueLine(
+			f.sumMetric(record.metric.Name()),
+			dp.Sum(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+
+		line = f.uintValueLine(
+			f.countMetric(record.metric.Name()),
+			dp.Count(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+	}
+
+	return lines
+}
+
+// doubleHistogram2Strings converts DoubleHistogram record to list of strings
+// for each data points: one per bound (with one for infinity) plus two for sum and count
```

Comment:
nitpick (similarly elsewhere?):

```suggestion
// doubleHistogram2Strings converts DoubleHistogram record to a list of strings,
// one for each data point: one per bound (with one for infinity) plus two for sum and count
```

Reply:
I would change that to `(n+1) where n is number of bounds plus two for sum and count`